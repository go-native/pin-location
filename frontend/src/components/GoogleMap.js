import React from "react"
import { compose, withProps } from "recompose"
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"

const Google = compose(
  withProps({
    googleMapURL: "https://maps.googleapis.com/maps/api/js?key=AIzaSyAIyiOIYzLHC1zAtUDITS8FeJ_I1NX_vc0&v=3.exp&libraries=geometry,drawing,places",
    loadingElement: <div style={{ height: `100%` }} />,
    containerElement: <div style={{ height: `400px` }} />,
    mapElement: <div style={{ height: `100%` }} />,
  }),
  withScriptjs,
  withGoogleMap
)((props) =>
  <GoogleMap
    defaultZoom={11}
    defaultCenter={{ lat: 52.511123, lng: 13.399605 }}
  >
    {props.addresses.map(address => <Marker key={address.address} position={{ lat: address.lat, lng: address.lng }} />)}
  </GoogleMap>
)

export default Google
