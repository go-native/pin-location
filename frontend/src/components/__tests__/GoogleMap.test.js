import GoogleMap from "./../GoogleMap"
import React from "react"
import { shallow } from "enzyme"

const addresses = [{
  id: "1544109127653",
  title: "Leipziger str.",
  address: "Leipziger strasse",
  lat: 52.511123,
  lng: 13.399605
},{
  id: "1544109127654",
  title: "Spandauer str.",
  address: "Spandauer strasse",
  lat: 52.511144,
  lng: 13.399655
}]

describe("GoogleMap", () => {
  it("should render properly", () => {
    const wrapper = shallow(<GoogleMap addresses={addresses} />)
    expect(wrapper).toMatchSnapshot()
  })
})
