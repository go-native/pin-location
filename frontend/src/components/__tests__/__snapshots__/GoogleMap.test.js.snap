// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap should render properly 1`] = `
<withScriptjs(withGoogleMap(Component))
  addresses={
    Array [
      Object {
        "address": "Leipziger strasse",
        "id": "1544109127653",
        "lat": 52.511123,
        "lng": 13.399605,
        "title": "Leipziger str.",
      },
      Object {
        "address": "Spandauer strasse",
        "id": "1544109127654",
        "lat": 52.511144,
        "lng": 13.399655,
        "title": "Spandauer str.",
      },
    ]
  }
  containerElement={
    <div
      style={
        Object {
          "height": "400px",
        }
      }
    />
  }
  googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyAIyiOIYzLHC1zAtUDITS8FeJ_I1NX_vc0&v=3.exp&libraries=geometry,drawing,places"
  loadingElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
  mapElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
/>
`;
